### DISCUSSION TITLE
Detections with stationary objects - both false and true positives

### DISCUSSION THREAD

**Discussion by Dinth** [blakeblackshear/frigate#9119] [General]
Hi. As im trying to use person/car detection for switching outside floodlights on and off, im having a big problem with the number of detections. The floodlights are on most of the night and as soon as they go off, in a few seconds they switch back on.
Today i thought i will easily fix that with stationary object detection, but the more i read about its behaviour here on the discussion forum, the more confused i am about how they actually work and if they can help me.

Looking at the history, the Frigate is constantly detecting stationary objects:
- true positives for example cars. A stationary, parked car can be detected over and over again throughout the night.
- false positives, for example trash bin, barbecue or a tree being detected as a person. This happens a lot on all my outside cameras - seems that each of the cameras picks one or two of such objects and stubbornly detects them as a person. 

Particularly with a person (and other living beings) i was hoping for a function which will stop detecting an object if its being stationary for lets say an hour - because nobody would stand there without a move for such a long time. Unfortunately it SEEMS to me now that this is not what stationary object detection is doing?

**Comment by NickM-27** [comment]
> Unfortunately it SEEMS to me now that this is not what stationary object detection is doing?

stationary object detection is the continuous detection of objects and recognizing that they are stationary. This allows frigate to not continuously track them but still know where they are. For example a car parked in the driveway - frigate knows this car is there but does not continuously look for it unless motion occurs around it.

There is a function for forcing stationary objects to be forgotten, but it is a last resort and there are other means to reduce false positives that are better.

**Reply by Dinth** [reply]
>This allows frigate to not continuously track them but still know where they are.

What is the practical benefit of this? Just less resource usage on EdgeTPU?

**Reply by NickM-27** [reply]
less resource usage on the tpu, but in general stationary object detection means frigate knows the car parked in your driveway is the same one from 6 hours ago and not a new car that just appeared. In frigate 0.13 it even marks the moments in time when the object started moving or became stationary so you can see in the timeline when (for example) a car driving by a house stopped for some time which may indicate abnormal behavior

**Reply by Dinth** [reply]
Is there a way to set different "stationary object detection" settings for different kinds of detection (car/person) or different cameras? 
What exactly is the benefit of adjusting interval?

**Reply by NickM-27** [reply]
what types of settings are you referring to?

**Reply by Dinth** [reply]
This one:
```
detect:
  stationary:
    interval: 0
    threshold: 50
```

**Reply by NickM-27** [reply]
Those can not be set per object.
